cfme/tests/configure/test_tag.py

```diff
@@ -107,6 +112,52 @@ def test_map_tagging_crud(appliance, category, soft_assert):
                                       .format(map_tag_entity.label))
 
 
+def test_tag_change_tag_name(provider, tag, request):
+    """
+    This test checks that tags don't disappear from the UI after their name (not displayed name) is
+    changed.
+
+    https://bugzilla.redhat.com/show_bug.cgi?id=1668730
+
+    Polarion:
+        assignee: anikifor
+        casecomponent: Configuration
+        caseimportance: high
+        initialEstimate: 1/8h
+        testSteps:
+            1. create a tag
+            2. assign the tag to some vm, observe the tag in Smart Management section of vm
+            3. change name of the tag
+            4. on VM screen: still the same tag in Smart Management section of vm
+    """
+    coll = provider.appliance.provider_based_collection(provider, coll_type='vms')
+    # need some VM to assign tags to, nothing specific is needed, so take the first one
+    vm = coll.all()[0]
+    vm.add_tag(tag)
+    request.addfinalizer(lambda: vm.remove_tag(tag))
+
+    # assert the tag is correctly assigned
+    vm_tags = vm.get_tags()
+    assert any(
+        tag.category.display_name == vm_tag.category.display_name and
+        tag.display_name == vm_tag.display_name
+        for vm_tag in vm_tags
+    ), "tag is not assigned"
+
+    # update the name of the tag
+    new_tag_name = '{}_{}'.format(tag.name, fauxfactory.gen_alphanumeric(4).lower())
+    tag.change_name(new_tag_name)
+
+    vm_tags = vm.get_tags()
+
+    # assert the tag was not changed in the UI
+    assert any(
+        tag.category.display_name == vm_tag.category.display_name and
+        tag.display_name == vm_tag.display_name
+        for vm_tag in vm_tags
+    ), 'tag is not assigned'
```

Comment:
BZ also states that vm should not show the tag when the tag is deleted .
Can we include that check too here ?

Reply:
@sshveta I don't see this in the BZ. The tag disappeared in the UI after the tag's name was changed, but it was a bug and was observer before the fix.